nbgitpuller/pull.py

```diff
@@ -43,23 +43,25 @@ def flush():
             raise subprocess.CalledProcessError(ret, cmd)
 
 class GitPuller(Configurable):
-    depth = Integer(None, allow_none=True, config=True,
-                    help="""
-                    Depth (ie, commit count) to which to perform a
-                    shallow git clone.
+    depth = Integer(
+        int(os.environ.get('NBGITPULLER_DEPTH', 1)),
+        config=True,
+        help="""
+        Depth (ie, commit count) to which to perform a
+        shallow git clone.
 
-                    If not set, disables shallow clones.
+        If not set, clones to depth 1.
 
-                    Defaults to reading from the NBGITPULLER_DEPTH
-                    environment variable.
-                    """)
+        Defaults to reading from the NBGITPULLER_DEPTH
+        environment variable."""
+    )
```

Comment:
- How to indicate that you want non-shallow clones?
- `allow_none` was removed, is ok?
- How does the default value specified here relate to the function below? I'm a bit confused about a default value is specified twice now, once in this traitlet declaration and once in the function below with a @default('depth') decorator.

Reply:
> How to indicate that you want non-shallow clones?

From a code inspection I found that it was by setting it to zero.

```python
        if self.depth and self.depth > 0:
            clone_args.extend(['--depth', str(self.depth)])
```

> allow_none was removed, is ok?

Yes I think so, it was needed for the default value to work.

> How does the default value specified here relate to the function below? I'm a bit confused about a default value is specified twice now, once in this traitlet declaration and once in the function below with a @default('depth') decorator.

This was resolved in another comment.